test/client/components/ShareTopicModal.test.js

```diff
@@ -0,0 +1,45 @@
+require('../support')
+import { mocks, helpers } from '../../support'
+import App from '../../../src/containers/App.js'
+import {
+  renderIntoDocument, findRenderedDOMComponentWithClass
+} from 'react-addons-test-utils'
+const { createElement } = helpers
+
+describe('ShareTopicModal', () => {
+  var store
+  const beta_access_code = 'foocode'
+  const slug = 'foomunity'
+  const tagName = 'bartag'
+  const currentCommunityId = 1
+
+  const setup = () => {
+    const component = createElement(App, {location: {}}, {store})
+    return renderIntoDocument(component).getWrappedInstance()
+  }
+
+  beforeEach(() => {
+    store = mocks.redux.store({
+      pending: {},
+      tagInvitationEditor: {},
+      tagsByCommunity: {},
+      tagPopover: {},
+      notifierMessages: [],
+      routing: {path: `/c/${slug}`},
+      people: {current: {id: 42}},
+      communities: {[slug]: {slug, beta_access_code, id: currentCommunityId}},
+      currentCommunityId,
+      showModal: {
+        show: 'share-tag',
+        params: {slug, tagName}
+      }
+    })
+  })
+
+  it('is rendered with correct join link', () => {
+    let appNode = setup()
+    let joinUrlDiv = findRenderedDOMComponentWithClass(appNode, 'join-url')
```

Comment:
I'm thinking we'll want to move all the data needed for App to render into it's own utility function once we use it again.